pyproject.toml

```diff
@@ -11,7 +11,7 @@ dynamic = ["version"]
 description = "pure-Python HistFactory implementation with tensors and autodiff"
 readme = "README.rst"
 license = { text = "Apache-2.0" }  # SPDX short identifier
-requires-python = ">=3.8"
+requires-python = ">=3.9"
```

Comment:
I'm personally fine with jumping up to Python 3.9. I think though as the `0.7.6` release is still on Python 3.7 that regardless of features we should make a `0.8.0` release after this that is the same as what we have in the `0.7.x` series but just with non-EOL Python support. That will make multiple things easier to maintain and Python 3.9 is not new at all so I don't see that as an issue.